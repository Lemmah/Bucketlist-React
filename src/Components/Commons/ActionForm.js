import React, { Component } from 'react'
import { Modal, Button, Form } from 'react-bootstrap'


class ActionForm extends Component {

  doAction = (e) => {
    let resourceName = this.refs.resourceName.value;
    let action = this.props.action.toLowerCase(),
        create = "create", add = "add";
    (action.includes(create) || action.includes(add) ?
      this.props.onCreateResource(resourceName)
      :
      this.props.onUpdateResource(resourceName)
    )
    e.preventDefault();
  }

  render () {
    return (
      <Modal 
        show={this.props.show}
        onHide={this.props.onHide}
        bsSize='small'
        aria-labelledby='contained-modal-title-lg'
      >
        <Form horizontal
          onSubmit={this.doAction.bind(this)}
        >
          <Modal.Header closeButton>
            <Modal.Title id='contained-modal-title-lg'>
              {this.props.action}
            </Modal.Title>
          </Modal.Header>
          <Modal.Body>
              <label>Name:</label>
              <input
                defaultValue={this.props.value}
                type="text" 
                className="form-control" 
                ref="resourceName"
              />
          </Modal.Body>
          <Modal.Footer>
            <Button
              bsStyle="danger"
              onClick={this.props.onHide}
            >
              Cancel
            </Button>
            <Button 
              bsStyle="success"
              type="submit"
            >
             {this.props.action.split(' ')[0]}
            </Button>
          </Modal.Footer>
        </Form>
      </Modal>
    )
  }
}

export default ActionForm
