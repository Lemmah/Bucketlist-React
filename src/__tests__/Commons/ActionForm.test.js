import React from 'react';
import renderer from 'react-test-renderer';
import ActionForm from '../../Components/Commons/ActionForm';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
configure({ adapter: new Adapter() });
import { shallow, mount, render } from 'enzyme';


describe('ActionForm test', () => {
	let action = 'Create Resource';
	const wrapper = shallow(
		<ActionForm
			action={action}
			show={true}
			onCreateResource={() => {}}
			onUpdateResource={() => {}} 
		/>
		);
	const instance = wrapper.instance();

  it('Displays ActionForm without Crashing', () => {
	const rendered = renderer.create(
		<ActionForm action={action}/>,
		);
	expect(rendered.toJSON()).toMatchSnapshot(); 
  });	

  it('Does Create Resource Action', () => {
  	action = "Create Bucketlist";
  	wrapper.find('#actionButton').simulate('click');
  })
});
