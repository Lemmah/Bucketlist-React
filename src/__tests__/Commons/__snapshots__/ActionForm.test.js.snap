// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionForm test Displays ActionForm without Crashing 1`] = `null`;
